cabal-install/Distribution/Client/CmdInstall.hs

```diff
@@ -560,18 +567,27 @@ installAction (configFlags, configExFlags, installFlags, haddockFlags, testFlags
     globalConfigFlag = projectConfigConfigFile (projectConfigShared cliConfig)
 
 -- | Install any built exe by symlinking/copying it
+-- we don't use BuildOutcomes because we also need the component names
 installExes :: Verbosity
             -> ProjectBaseContext
             -> ProjectBuildContext
+            -> Platform
             -> Compiler
             -> ClientInstallFlags
             -> IO ()
-installExes verbosity baseCtx buildCtx compiler clientInstallFlags = do
-  -- XXX The comment in InstallSymlink.hs (pkgBinDir) says this is too naive (and it is)
-  let mkPkgBinDir = (</> "bin") .
-                    storePackageDirectory
-                       (cabalStoreDirLayout $ cabalDirLayout baseCtx)
-                       (compilerId compiler)
+installExes verbosity baseCtx buildCtx platform compiler
+            clientInstallFlags = do
+  let storeDirLayout = cabalStoreDirLayout $ cabalDirLayout baseCtx
+  let mkPkgBinDir :: UnitId -> FilePath
+      mkPkgBinDir = InstallDirs.bindir .
+                    storePackageInstallDirs'
+                      storeDirLayout
+                      (compilerId compiler)
+      mkExeName :: UnqualComponentName -> FilePath
+      mkExeName exe = unUnqualComponentName exe <> ext
+        where
+          ext = case exeExtension platform of "" -> ""
```

Comment:
Hmm, can't we use `<.>` from `System.FilePath` here?

Reply:
Didn't know about that function, thanks